src/modules/todolist.js

```diff
@@ -0,0 +1,134 @@
+    // Update the list in local storage....😎😊😎
+    const checkbox = event.target;
+    const taskElm = checkbox.parentNode.parentNode;
+    const listindex = Array.prototype.indexOf.call(listHolder.children, taskElm);
+    listArray[listindex].completed = checkbox.checked;
+    localStorage.setItem('listArray', JSON.stringify(listArray));
+
+    if (event.target.type === 'checkbox' && event.target.checked) {
+      event.target.nextElementSibling.style.textDecoration = 'line-through';
+    } else if (event.target.type === 'checkbox' && !event.target.checked) {
+      event.target.nextElementSibling.style.textDecoration = 'none';
+    }
+  });
+
+  if (localStorage.getItem('listArray')) {
+    listArray = JSON.parse(localStorage.getItem('listArray'));
+    UI.displayData();
+  }
+
+  listHolder.addEventListener('click', (event) => {
+    if (event.target.classList.contains('fa-ellipsis-v')) {
+      event.target.classList.remove('fa-ellipsis-v');
+      event.target.classList.add('fa-trash');
+      event.target.parentElement.parentElement.style.backgroundColor = '#b4d5fe';
+      const targeted = event.target.parentNode.previousElementSibling.lastElementChild;
+      targeted.contentEditable = 'true';
+    } else if (event.target.classList.contains('fa-trash')) {
+      // Get the index of the element to be deleted....😎😎😎😎
+      const trash = event.target;
+      const taskElm = trash.parentNode.parentNode;
+      const listindex = Array.prototype.indexOf.call(listHolder.children, taskElm);
+
+      // Remove the element from the listArray
+      listArray.splice(listindex, 1);
+
+      // Update the local storage with the updated listArray......😎😎😎😎
+      localStorage.setItem('listArray', JSON.stringify(listArray));
+
+      // Remove the element from the DOM....😎😊😉😉
+      event.target.parentElement.parentElement.remove();
+    }
+  });
+
+  listHolder.addEventListener('input', (e) => {
+    if (e.target.closest('#discription')) {
+      const inputText = e.target.closest('#discription');
+      const taskElm = inputText.parentNode.parentNode;
+      const index = Array.prototype.indexOf.call(listHolder.children, taskElm);
+      listArray[index].discription = e.target.textContent;
+
+      localStorage.setItem('listArray', JSON.stringify(listArray));
+    }
+  });
+
+  window.addEventListener('load', () => {
+    if (localStorage.getItem('listArray')) {
+      listArray = JSON.parse(localStorage.getItem('listArray'));
+      UI.displayData();
+    }
+  });
+
+  if (localStorage.getItem('listArray')) {
+    listArray = JSON.parse(localStorage.getItem('listArray'));
+    UI.displayData();
+  }
+  // delete the checked list from the task list....😎😊😉
+
+  buttonClearAll.addEventListener('click', () => {
+    listArray = listArray.filter((item) => item.completed !== true);
+    localStorage.setItem('listArray', JSON.stringify(listArray));
+
+    UI.displayData();
+  });
+};
+export default todolist;
```

Comment:
- [ ] [OPTIONAL] You have done a great job with the descriptive comments @ismailmunyentwari9 👍
However, we can still limit them in order to maintain a clean code base.

Reply:
Thank you sir..
I will keep that in mind